### ISSUE TITLE
Capture the SHAs of test repos used

### ISSUE THREAD

**Issue by llxia** [AdoptOpenJDK/openjdk-tests#1218]
We need to capture the used test repo SHAs. The solution should support both local runs and Jenkins builds (both archive into Jenkins master and Artifactory).

- A script `getSHA.sh` uses git commands to get sha in the local git dir. It outputs into the console and stores the info into a user-provided file. It takes two parameters: 
 -> `REPO_DIR` - local git repo directory
 -> `OUTPUT_FILE` - file to write the sha info to. Default is to `TestConfig/SHA.txt`

```
echo "================================================"; 
echo "timestamp: $(timestamp)"; 
echo "repo dir: $REPO_DIR"; 
echo "git repo: "; git remote show origin -n | grep "Fetch URL:"; 
echo "sha:"; 
git rev-parse HEAD;
```
Sample output:
```
================================================
timestamp: 20190715-072807
repo dir: /cygdrive/c/Users/jenkins/workspace/Grinder_Sandbox/openjdk-tests
git repo: 
  Fetch URL: https://github.com/llxia/openjdk-tests.git
sha:
7cb568b0aa71f0a13930edc4e5eb7d8316d31382
================================================
timestamp: 20190715-072807
repo dir: /cygdrive/c/Users/jenkins/workspace/Grinder_Sandbox/openjdk-tests/openj9
git repo: 
  Fetch URL: https://github.com/llxia/openj9.git
sha:
4ebbc8dbccab80e0ce9fbfbf298298df9ae7af19
```
- `getSHA.sh` will be invoked in places that run git clone (i.e., get.sh, any build.xml, etc)

- In the top-level ant `TestConfig/scripts/build_test.xml`, we will create an ant macrodef called checkGitRepoSha. This macrodef calls `getSHA.sh`.

- In each `build.xml`, checkGitRepoSha can be called with required parameters after each git clone
```
<checkGitRepoSha repoDir="${SYSTEMTEST_ROOT}/openj9-systemtest" />
```

- In JenkinsfileBase, `TestConfig/SHA.txt` will be archived regardless of the build result. If `ARTIFACTORY_SERVER ` is provided, we will try to store the file on the `ARTIFACTORY_SERVER ` server.

Related to https://github.com/AdoptOpenJDK/openjdk-tests/issues/1169

**Comment by llxia** [comment]
PR https://github.com/eclipse/openj9/pull/6460
PR https://github.com/AdoptOpenJDK/openjdk-tests/pull/1219

**Comment by smlambert** [comment]
Closed via #1219 and https://github.com/eclipse/openj9/pull/6460
